src/plugins/workspace/public/components/workspace_form/workspace_privacy_setting_panel.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiCheckableCard,
+  EuiCheckbox,
+  EuiCompressedFormRow,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import {
+  optionIdToWorkspacePermissionModesMap,
+  privacyTypeEditDescription,
+  privacyTypeEditTitle,
+  privacyTypePrivateDescription,
+  privacyTypePrivateTitle,
+  privacyTypeViewDescription,
+  privacyTypeViewTitle,
+  WorkspacePermissionItemType,
+  WorkspacePrivacyItemType,
+} from './constants';
+import { WorkspacePermissionSetting } from './types';
+import { PermissionModeId } from '../../../../../core/public';
+import './workspace_privacy_setting.scss';
+
+export interface WorkspacePrivacySettingProps {
+  onPermissionChange: (
+    value: Array<Pick<WorkspacePermissionSetting, 'id'> & Partial<WorkspacePermissionSetting>>
+  ) => void;
+  permissionSettings: Array<
+    Pick<WorkspacePermissionSetting, 'id'> & Partial<WorkspacePermissionSetting>
+  >;
+  goToCollaborators: boolean;
+  onGoToCollaboratorsChange: (value: boolean) => void;
+}
+
+export const WorkspacePrivacySettingPanel = ({
+  onPermissionChange,
+  permissionSettings,
+  goToCollaborators,
+  onGoToCollaboratorsChange,
+}: WorkspacePrivacySettingProps) => {
+  const [privacyType, setPrivacyType] = useState(WorkspacePrivacyItemType.PrivateToCollaborators);
```

Comment:
Can we move `privacyType` to `useWorkspaceForm`, then we can share with create and detail page. Then we don't need to pass `permissionSettings` and `onPermissionChange` in this component. And the privacyType should calculate from `permissionSettings` in the initial rendering.